### DISCUSSION TITLE
Cancel macro recording

### DISCUSSION THREAD

**Discussion by cool-RR** [saulpw/visidata#1810] [Q&A] [answered]
I accidentally pressed `m` and started recording a macro. How do I stop this? I was surprised there isn't a command for that.

**Comment by anjakefala** [comment]
Hi @cool-RR!

The current available method is to press `m` again, and then `Ctrl+C`, which is not ideal. It should be straightforward enough for me to add a command that you can bind!

**Comment by geekscrapy** [comment, marked as answer]
Just wanted to add I've been bitten before by this. I couldn't find a way to cancel so ended up with a macro of 100s of steps for the rest of the session 🤣
Maybe when m is pressed for the second time, the prompt could show "macro name (ctrl+c to cancel)"?

**Reply by saulpw** [reply]
I like this idea, let's do this and if we still feel like we need a separate command, we can add it also.

**Reply by geekscrapy** [reply]
Might best best if it was explicit "macro name (ctrl+c to cancel macro recording)", or something similar

**Reply by anjakefala** [reply]
I added it!

**Reply by cool-RR** [reply]
Agreed, this is good.
